app/models/remote_record.rb

```diff
@@ -0,0 +1,87 @@
+# frozen_string_literal: true
+class RemoteRecord
+  def self.retrieve(source_metadata_identifier)
+    if PulMetadataServices::Client.bibdata?(source_metadata_identifier)
+      new(source_metadata_identifier)
+    else
+      PulfaRecord.new(source_metadata_identifier)
+    end
+  end
+
+  class PulfaRecord
+    attr_reader :source_metadata_identifier
+    def initialize(source_metadata_identifier)
+      @source_metadata_identifier = source_metadata_identifier
+    end
+
+    delegate :attributes, to: :client_result
+
+    def success?
+      client_result.source.strip.present?
+    end
+
+    def client_result
+      @client_result ||= PulMetadataServices::Client.retrieve(source_metadata_identifier)
+    end
+  end
+
+  attr_reader :source_metadata_identifier
+  delegate :success?, to: :jsonld_request
+  def initialize(source_metadata_identifier)
+    @source_metadata_identifier = source_metadata_identifier
+  end
+
+  def attributes
+    JSONLDBuilder.for(jsonld).result
+  end
+
+  private
+
+    def jsonld_request
+      @jsonld_request ||= Faraday.get("https://bibdata.princeton.edu/bibliographic/#{source_metadata_identifier}/jsonld")
```

Comment:
TODO: we shouldn't be generating bibdata and pulfa URLs in our apps — I've added a ticket to pul_metadata_services for this: https://github.com/pulibrary/pul_metadata_services/issues/18